### PR TITLE
sensors: move to uORB::Subscription and reduce CONFIG_NFILE_DESCRIPTORS 20 -> 15

### PR DESCRIPTION
**Pull request by dagar**
~~Requires https://github.com/PX4/Firmware/pull/14096~~ Done

Saves more memory and cpu.

### CODE REVIEW

**Comment by dannyfpv** [comment]
Tested on pixhawk4 v5 f-450
Indoor Flight Test
Modes Tested
Stabilized Mode: Good.

Procedure
Flashed the firmware, Arm and Take off in stabilized mode, right off the bat we noticed oscillations. We landed the vehicle and tried to calibrate the accelerator, NO accelerator on board message. Image below 
Screen Shot 2020-03-13 at 12 24 02 PM


Notes
No issues noted, good flight in general.

log:
https://review.px4.io/plot_app?log=2990cd6a-8d83-4683-8183-ac789d800f70

**Comment by jorge789** [comment]
Tested on PixRacer V4
Indoor Flight Test
Modes Tested
Stabilized Mode: Good.

Procedure
Flashed the firmware, Arm and Take off in stabilized mode.We landed the vehicle and tried to calibrate the accelerator, NO accelerator on board message. Image below

Log:  https://review.px4.io/plot_app?log=1840f9db-e32e-4452-9a37-0c4a3bc6d9dc

**Comment by Junkim3DR** [comment]
### Tested on NXP FMUK66 v3

**Indoor Flight**

**Modes Tested**
- Stabilized Mode: Good.

**Note**
Slight vibrations noted.

**Log**
https://review.px4.io/plot_app?log=fadbc945-00d1-4913-a912-a624fb4f1af6

File: src/modules/sensors/voted_sensors_update.cpp [hunk 1]
```diff
@@ -483,27 +466,15 @@ void VotedSensorsUpdate::accelPoll(struct sensor_combined_s &raw)
...
-			}
+		if (_accel.enabled[uorb_index] && _accel.subscription[uorb_index].update(&accel_report)) {
 
 			// First publication with data
 			if (_accel.priority[uorb_index] == 0) {
-				int32_t priority = 0;
-				orb_priority(_accel.subscription[uorb_index], &priority);
-				_accel.priority[uorb_index] = (uint8_t)priority;
+				ORB_PRIO priority = _accel.subscription[uorb_index].get_priority();
+				_accel.priority[uorb_index] = priority;
```
Comment (bkueng) [inline src/modules/sensors/voted_sensors_update.cpp]: You could merge this into one line.

**Comment by dagar** [comment]
It looks like we're not quite ready to go down to 10 FDs. I'll keep it higher for now and take another pass later after moving the calibration code to uORB::Subscription.

File: src/modules/sensors/voted_sensors_update.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/sensors/voted_sensors_update.cpp]: Fixed.

**Comment by dagar** [comment]
During accel calibration.

``` Console
 PID COMMAND                   CPU(ms) CPU(%)  USED/STACK PRIO(BASE) STATE FD
   0 Idle Task                  277352 53.437   244/  512   0 (  0)  READY  3
   1 hpwork                          0  0.000   344/ 1260 249 (249)  w:sig  3
   2 lpwork                          4  0.000   976/ 1580  50 ( 50)  w:sig  3
   3 init                         1156  0.000  1848/ 2604 100 (100)  w:sem  3
   4 wq:manager                      0  0.000   408/ 1252 255 (255)  w:sem  4
 176 mavlink_rcv_if0              1426  0.261  2664/ 3908 175 (175)  READY  4
  15 wq:hp_default                3415  0.696   972/ 1900 241 (241)  READY  4
  33 wq:SPI1                     32198  6.353  1136/ 2364 253 (253)  w:sem  4
  19 wq:lp_default                 154  0.087   816/ 1700 205 (205)  w:sem  4
  35 wq:SPI5                     54413 11.401  1112/ 2364 249 (249)  w:sem  4
  38 wq:SPI2                       869  0.174   944/ 2364 252 (252)  w:sem  4
  40 wq:I2C1                      5041  0.957   828/ 1396 246 (246)  READY  4
  61 wq:att_pos_ctrl             51600 10.356  5016/ 7196 242 (242)  READY  4
  62 wq:rate_ctrl                14723  2.872  1080/ 1596 255 (255)  w:sem  4
  70 commander                    4691  0.957  1576/ 3212 140 (140)  READY 11
  71 commander_low_prio            248  0.522  2432/ 2996  50 ( 50)  w:sem 11
  77 mavlink_if0                 19920  4.003  1832/ 2572 100 (100)  READY  4
 106 gps                           299  0.087  1016/ 1676 208 (208)  w:sem  4
 127 mavlink_if1                  6136  1.218  1832/ 2484 100 (100)  READY  4
 133 mavlink_rcv_if1              1570  0.348  2640/ 3908 175 (175)  w:sem  4
 145 wq:UART3                     2161  0.435   672/ 1396 236 (236)  w:sem  4
 150 navigator                     409  0.087   868/ 1764 105 (105)  w:sem  5
 171 logger                       1619  0.261  1688/ 3644 233 (233)  w:sem  3
 172 log_writer_file                 0  0.000   368/ 1164  60 ( 60)  w:sem  3
 184 top                            92  2.697  1560/ 2028 248 (248)  RUN    3

```

**Pull request merged**